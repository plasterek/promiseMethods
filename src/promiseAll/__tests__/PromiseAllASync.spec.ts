import { promiseAllASync } from "../PromiseAllASync.service";

describe("PromiseAllAsync function", () => {
  const succces: string = "resolved";
  const failure: string = "rejected";
  const resolvedPromise: Promise<string> = Promise.resolve(succces);
  const rejectedPromise: Promise<string> = Promise.reject(failure);

  describe("When every promise in given array is resolved", () => {
    it("It returns Promise with array of given promise results", async () => {
      //given
      const promiseArray: Promise<string>[] = [resolvedPromise, resolvedPromise];
      //when
      const results: string[] = await promiseAllASync(promiseArray);
      //then
      expect.assertions(1);
      expect(results).toMatchObject([succces, succces]);
    });
  });

  describe("When any promise rejects in given array", () => {
    it("It returns Promise with rejected value", async () => {
      //given
      const promiseArray: Promise<string>[] = [resolvedPromise, rejectedPromise];
      //when
      try {
        const result: string[] = await promiseAllASync(promiseArray);
        //then
      } catch (err) {
        expect.assertions(1);
        expect(err).toBe(failure);
      }
    });
  });

  describe("When given array is empty", () => {
    it("It returns Promise with empty array", async () => {
      //when
      const results: string[] = await promiseAllASync([]);
      //then
      expect.assertions(1);
      expect(results).toMatchObject([]);
    });
  });
});
