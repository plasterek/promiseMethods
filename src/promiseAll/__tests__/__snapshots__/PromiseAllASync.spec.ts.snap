// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromiseAllAsync function When any promise rejects in given array It returns Promise with rejected value 1`] = `"rejected"`;

exports[`PromiseAllAsync function When every promise in given array is resolved It returns Promise with array of given promise results 1`] = `
[
  "resolved",
  "resolved",
]
`;

exports[`PromiseAllAsync function When given array is empty It returns Promise with empty array 1`] = `[]`;
